### ISSUE TITLE
`no-empty-sections` (`MD082`) - option to allow empty section when the next heading is a subheading

### ISSUE THREAD

**Issue by DetachHead** [rvben/rumdl#718]
### Problem or motivation

i personally don't consider it an issue for a section to contain no content, when followed by subheadings, because those subheadings should be considered part of the parent heading's content

### Proposed solution

an option for `no-empty-sections` to allow "empty" sections as long as it contains subheadings

### Feature category

Improve existing rule

### Example markdown

```markdown
<!-- this should be fine: -->
# H1

## H2

content...


<!-- but this isn't: -->
# H1

# H1

content...
```

### Contribution

- [ ] I would be willing to submit a PR to implement this feature

**Comment by rvben** [comment]
Shipped in v0.2.48:

```toml
[MD082]
allow-parent-headings = true
```

With it on, an empty section is accepted when the next heading is a subheading, so the first half of your example is clean. The second half (`# H1` followed by another `# H1`) is still reported. Default is `false`, so nothing changes for existing configurations.

Thanks @DetachHead, the two-part example made the boundary unambiguous.
